In [4]:
import numpy as np
import pandas as pd
dt=pd.read_csv('data.csv')
dt.head()

,SALESPERSON,Old Scheme (in thousands),New Scheme (in thousands)
0,1,57,62
1,2,103,122
2,3,59,54
3,4,75,82
4,5,84,84


### 1. Find the mean of old scheme and new scheme column.

In [10]:
dt.describe()
dt.mean()
print("Old Scheme (in thousands)=",dt["Old Scheme (in thousands)"].mean())
print("New Scheme (in thousands)=",dt["New Scheme (in thousands)"].mean())

Old Scheme (in thousands)= 68.03333333333333
New Scheme (in thousands)= 72.03333333333333


In [25]:
import numpy as np
import pandas as pd
from scipy.stats import ttest_1samp, wilcoxon, ttest_ind, mannwhitneyu, levene, shapiro, f_oneway
from statsmodels.stats.power import ttest_power
from statsmodels.stats.multicomp import pairwise_tukeyhsd, MultiComparison
import matplotlib.pyplot as plt
import seaborn as sns
import os
%matplotlib inline
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

In [30]:
shapiro(dt.iloc[:,2])
shapiro(dt["New Scheme (in thousands)"])

(0.9687567353248596, 0.5057420134544373)

In [39]:
#Can't reject the null Hypothesis means Data for "New Scheme (in thousands)" is Normally Distributed.
shapiro(dt.iloc[:,1])
shapiro(dt["Old Scheme (in thousands)"])

(0.9885101914405823, 0.9813658595085144)

In [ ]:
#Can't reject the null Hypothesis means Data for "Old Scheme (in thousands)" is Normally Distributed.

In [33]:
#t_statistic, p_value = ttest_1samp(dt["New Scheme (in thousands)"], dt["New Scheme (in thousands)"].mean())
t_statistic, p_value = ttest_1samp(dt["New Scheme (in thousands)"], 0)
print(t_statistic, p_value)

16.396656129293135 3.307814709155835e-16


### 2. Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs? (10 points)

In [ ]:
# twosample_results = scipy.stats.ttest_ind(dt["Old Scheme (in thousands)"], dt["New Scheme (in thousands)"])
t_statistic, p_value = ttest_ind(dt["Old Scheme (in thousands)"], dt["New Scheme (in thousands)"])
print(t_statistic, p_value)


In [37]:
print ("two-sample t-test p-value=", p_value)

two-sample t-test p-value= 0.49063515686248105


In [38]:
#u, p_value = mannwhitneyu(dt["Old Scheme (in thousands)"], dt["New Scheme (in thousands)"])
#print ("two-sample wilcoxon-test p-value=", p_value)

two-sample wilcoxon-test p-value= 0.24815395764270332


### 3. What conclusion does the test (p-value) lead to?

## the null hypothesis is that the means of both groups "Old Scheme (in thousands)" and "New Scheme (in thousands)" are the same
#As two-sample t-test p-value= 0.49063515686248105 is around 49% above 5% that is we can't reject the NULL hypothesis, conclude that Means of Both the groups ("Old Scheme (in thousands)" and "New Scheme (in thousands)") are not same.


In [40]:
dt.describe()

,SALESPERSON,Old Scheme (in thousands),New Scheme (in thousands)
count,30.000000,30.000000,30.000000
mean,15.500000,68.033333,72.033333
std,8.803408,20.455980,24.062395
min,1.000000,28.000000,32.000000
25%,8.250000,54.000000,55.000000
50%,15.500000,67.000000,74.000000
75%,22.750000,81.500000,85.750000
max,30.000000,110.000000,122.000000


### 4.a The probability of a type 1 error?

In [ ]:
# Ans: Probability of Type I error = significant level = 0.05 or 5%

### 4.b What is the p- value of the hypothesis test if we test for a difference of $5000?

In [46]:
# H1: (Mean of "New Scheme (in thousands)") - (Mean of "Old Scheme (in thousands)") >5000 (alternative Hypothesis)
# H0: (Mean of "New Scheme (in thousands)") - (Mean of "Old Scheme (in thousands)")<=5000 (NULL Hypothesis)
#t_statistic, p_value = ttest_1samp(dt["Old Scheme (in thousands)"] - dt["New Scheme (in thousands)"], 0)
#print(t_statistic, p_value)
z_statistic, p_value = wilcoxon(dt["Old Scheme (in thousands)"] - dt["New Scheme (in thousands)"])

print ("paired wilcoxon-test p-value=", p_value)
#levene(dt["Old Scheme (in thousands)"], dt["New Scheme (in thousands)"])
#import scipy.stats as stats
#t, p = stats.ttest_rel(dt["Old Scheme (in thousands)"], dt["New Scheme (in thousands)"])
#t, p

paired wilcoxon-test p-value= 0.06116952762758769


In [ ]:
# Ans: Since p-value (=0.06116952762758769) is higher than 0.05, we accept (fail to reject) NULL hypothesis. The new scheme has NOT significantly raised outputs.

### 4.c Power of the test

In [53]:
# Calculating Power of Test
# Compute the difference in Means between 2 sample means and divide by pooled Standard Deviation 
# number of Observations/tuples
# Set the alpha value to 0.05 and alternative values 'two-sided' , 'larger' , 'smaller'
pre = dt["Old Scheme (in thousands)"]
post= dt["New Scheme (in thousands)"]
#dt.count
(np.mean(pre) - np.mean(post)) / np.sqrt(((30-1)*np.var(pre)+(30-1)*np.var(post)) / 30+30-2)\


-0.12909555513169643

In [55]:
print(ttest_power(0.129, nobs=30, alpha=0.05, alternative='two-sided'))

0.10498428747328836
